tools/jerry_heaplimit_measure.py

```diff
@@ -0,0 +1,138 @@
+import json
+import os
+import subprocess
+import sys
+
+from subprocess import Popen, PIPE, STDOUT
+
+
+BIN_IOTJS_DEBUG = os.path.join(os.getcwd(),'build/x86_64-linux/debug/bin/iotjs')
+BIN_IOTJS_RELEASE = os.path.join(os.getcwd(),
+                                        'build/x86_64-linux/release/bin/iotjs')
+FLAG_BUILDTYPE = '--buildtype'
+FLAG_CLEAN = '--clean'
+FLAG_NO_TEST = '--no-check-test'
+FLAG_HEAPLIMIT = '--jerry-heaplimit'
+IOTJS_BUILDER = os.path.join(os.getcwd(), 'tools/build.py')
+TEST_DIR = os.path.join(os.getcwd(), 'test')
+
+
+def get_args():
+    """ Parse input arguments. """
+    parser = argparse.ArgumentParser(description='Finds the smallest possible \
+                    size of JerryHeap without failing to run the given js file')
+    parser.add_argument('testfile')
+    parser.add_argument('--heapsize', type = int, default = 256,
+                         help = 'Set the limit of the first heapsize')
+    parser.add_argument('--buildtype', type = str, default = 'debug',
+                                        help = 'Set the jerry buildtype')
+
+    script_args = parser.parse_args()
+
+    return script_args
+
+
+def is_exe(fpath):
+    """ Check whether the file is exist and executable. """
+    return os.path.isfile(fpath)
+
+
+def init(opts):
+    if not is_exe(IOTJS_BUILDER):
+        sys.exit("Builder script isn't a file!")
+    if not is_exe(opts.testfile):
+        sys.exit("Testfile isn't a file!")
+
+
+def median(low, hi):
+    """ Get the middle value of two numbers """
+    if (low + hi) % 2:
+        return (low + hi) / 2
+    else:
+        return (low + hi + 1) / 2
+
+
+def build_bin(heapsize, opts):
+    """ Run tools/build.py script """
+    buildtype = opts.buildtype
+    hsize = str(heapsize)
+    subprocess.call([IOTJS_BUILDER, FLAG_CLEAN, FLAG_NO_TEST,
+                FLAG_BUILDTYPE, buildtype, FLAG_HEAPLIMIT, hsize])
+
+
+def get_output(cwd, cmd):
+    """ Run the given command and return with the output. """
+    process = Popen(cmd, stdout=PIPE, stderr=STDOUT, cwd=cwd)
+
+    output = process.communicate()[0]
+    exitcode = process.returncode
+
+    return output, exitcode
+
+
+def run_test(opts):
+    """ Run the testfile for output and exitcode. """
+    try:
+        testfile = os.path.abspath(opts.testfile)
+        if opts.buildtype in ['debug']:
+            run_cmd = [BIN_IOTJS_DEBUG, testfile]
+        else:
+            run_cmd = [BIN_IOTJS_RELEASE, testfile]
+        output, ret = get_output(TEST_DIR, run_cmd)
+    except subprocess.CalledProcessError as err:
+        output = err.output
+        ret = err.returncode
+
+    return opts.testfile, output, ret
+
+
+def heap_limit(opts):
+    """ Find the minimal size of jerryheap to pass """
+    goodheap = int(opts.heapsize)
+    lowheap = int(0)
+    hiheap = int(opts.heapsize)
+
+    while lowheap < hiheap:
+        build_bin(hiheap, opts)
+
+        if opts.buildtype in ['debug']:
+            is_exe(BIN_IOTJS_DEBUG)
```

Comment:
You should handle the case when there is no binary. You can add an assert for this line.